drivers/flash/flash_stm32_qspi.c

```diff
@@ -886,6 +887,61 @@ static void flash_stm32_qspi_pages_layout(const struct device *dev,
 }
 #endif
 
+#if defined(CONFIG_FLASH_EX_OP_ENABLED)
+#if defined(CONFIG_FLASH_STM32_QSPI_GENERIC_READ)
+static int flash_stm32_qspi_generic_read(const struct device *dev, QSPI_CommandTypeDef *cmd,
+					 void *out)
+{
+	int ret;
+
+	qspi_lock_thread(dev);
+
+	ret = qspi_read_access(dev, cmd, out, cmd->NbData);
+
+	qspi_unlock_thread(dev);
+
+	return ret;
+}
+#endif /* CONFIG_FLASH_STM32_QSPI_GENERIC_READ */
+
+#if defined(CONFIG_FLASH_STM32_QSPI_GENERIC_WRITE)
+static int flash_stm32_qspi_generic_write(const struct device *dev, QSPI_CommandTypeDef *cmd,
+					  void *in)
+{
+	int ret;
+
+	qspi_lock_thread(dev);
+
+	ret = qspi_write_access(dev, cmd, in, cmd->NbData);
+
+	qspi_unlock_thread(dev);
+
+	return ret;
+}
+#endif /* CONFIG_FLASH_STM32_QSPI_GENERIC_READ */
+
+static int flash_stm32_qspi_ex_op(const struct device *dev, uint16_t code, const uintptr_t cmd,
+				  void *data)
+{
+	int rv = -ENOTSUP;
+
```

Comment:
I think you need to consider cases where userland with `CONFIG_USERSPACE=y` gets this API called.
See *flash_stm32_ex_op.c* as example.

Reply:
I added support the userland case. It is the first time I had to manage this but the code seems quite straightforward to me. Let me know if anything could be done better.